### ISSUE TITLE
Q_extraSD_Acoustic_Survey(2) no longer output in keyposteriors.csv file

### ISSUE THREAD

**Issue by cgrandin** [cgrandin/hake-assessment#301]
Breaks the mcmc diagnostic plot showing this parameter (Fig. 17 from last year's assessment.)

**Comment by cgrandin** [comment]
I temporarily commented out the code to make this plot, so leave this open until it is uncommented

**Comment by taylori** [comment]
Looks like with the change from SS version 3.24 to 3.30, the parameter label format changed 
I made I fixed it in this commit https://github.com/taylori/hake-assessment/commit/0bf885aec1f2fed224c5fd00d8062a6d5999a5b7
but haven't compiled yet to test it.

@cgrandin or @andrew-edwards, if you have a chance, could you add something about the workflow for branches forked from branches to the README? I was able to get a matching branch to Chris's, but I'm not quite sure how I got there or what the right steps are in general.

**Comment by andrew-edwards** [comment]
Chris and I have been in a workshop all day (and he just left). From the Network Graph it looks like you fetched Chris's v33 branch, which is ahead of master (so you fetched and merged v33 not master).  You can create another branch from that if you like with cb <newbranch>  , which will just branch off again.
I presume Chris hadn't merged his stuff back into master because it's still ongoing [kind of the point of branches - if you're working on something that might break the code you don't want to mess up other people].  
We have a few branching commands here: https://github.com/andrew-edwards/git-course 
Not sure if this helps - I can add to the readme if necessary.

**Comment by taylori** [comment]
Thanks Andy.
I was guessing that Chris wasn't going to merge the branch until he got everything working with ss 3.30 and I just wanted to contribute to that effort and figured that it made sense to also use a branch (but maybe that isn't necessary). I ended up using the GitHub desktop application but would prefer to know better how the remote, fetch, and merge command line commands work when you want to fetch from another users branch rather than master.

**Comment by andrew-edwards** [comment]
Ian - did you create a new branch with the same name as Chris's? That might be interesting..... but should all work out I guess.

**Comment by taylori** [comment]
I figured that instead of cgrandin/master and taylori/master, we could be working in parallel on cgrandin/v33 and taylori/v33, but if that causes confusion it seems that renaming branches is not too hard.

**Comment by andrew-edwards** [comment]
Yes, you're working in parallel on your own versions of v33, but since you fetched and merged Chris's version in 80da7b908a  then you're just building on his work. So essentially the same idea as if you're both working on master, except the v33 branch is working on something specific that may break the code (and so others can still use master until you merge back in). Hope that makes sense. Thinking I won't try anything until everything's merged back in.

**Comment by cgrandin** [comment]
Just to follow up on this, the actual value is no
longer in the keyposteriors.csv file:
![capture](https://user-images.githubusercontent.com/4551051/33848863-def99994-de63-11e7-8dd8-22847861ff40.PNG)

**Comment by taylori** [comment]
Thanks for checking. I'm finally trying to install proTeXt now to see if that allows me to compile again instead of continuing to push commits without fully testing them in the compiling step. My previous commit failed to account for R changing "Q_extraSD_Acoustic_Survey(2)" to "Q_extraSD_Acoustic_Survey.2." but I think the functions in question do partial matching, so it should now work with just "Q_extraSD" as updated in commit 73627bb.

The writing and then reading again of the keyposteriors.csv and nuisanceposteriors.csv files is more complex than necessary because it follows the convoluted logic the r4ss SSgetMCMC function. Those same quantities are in model$mcmc so we could skip the file write/read steps by just subsetting model$mcmc using the same logic. But hopefully the commit I just made will succeed such that it's not necessary to take the time to mess with this stuff further.

**Comment by cgrandin** [comment]
The reason I had `Q_extraSD_?[0-9]?_Acoustic_Survey` was so that it would match both 3.24 and 3.30 versions. It does partial so you don't need the (2) at the end.

I did it that way just so I could change the base model from last year's to this years 3.30 version and still leave all the other ones in there so we didn't have to comment a whole bunch of stuff out, then uncomment again later. So the way V33 branch works now is - you drop `03_2017base_3.30_tv_pars` model into the models directory along with all last year's, then run the build() command to run the new code to load everything for 3.30, then compile the document.

I hope this was a good way of getting it going!

**Comment by taylori** [comment]
My understanding of the regular expressions was too weak to get what the question marks were doing to understand that this could match both types, which is definitely useful. In that case, I'm not sure what the problem is that caused the original issue.
